src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,191 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head = null;
+    private Node<T> tail = null;
+    private int size = 0;
+
+    static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T item, Node<T> next) {
+            this.item = item;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            Node<T> newNode = new Node<T>(null, value, null);
+            head = newNode;
+            tail = head;
+        } else {
+            Node<T> newNode = new Node<T>(tail, value, null);
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> currentNode;
+        if (index == 0) {
+            if (head == null) {
+                Node<T> newNode = new Node<T>(null, value, null);
+                head = newNode;
+                tail = head;
+            } else {
+                currentNode = new Node<>(null, value, head);
+                head.prev = currentNode;
+                head = currentNode;
+            }
+        } else if (index == size) {
+            add(value);
+            return;
+        } else if (index == (size - 1)) {
+            if (checkIndex(index)) {
+                currentNode = new Node<>(tail.prev, value, tail);
+                tail.prev.next = currentNode;
+                tail.prev = currentNode;
+            }
+        } else {
+            currentNode = searchNodeByIndex(index);
+            Node<T> newNode = new Node<T>(currentNode.prev, value, currentNode);
+            newNode.prev.next = newNode;
+            currentNode.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return searchNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> currentNode;
+        T e = null;
+        currentNode = searchNodeByIndex(index);
+        e = currentNode.item;
+        currentNode.item = value;
+        return e;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T deleteElement = null;
+        if (checkIndex(index)) {
+            deleteElement = (index == 0) ? unlinkFirst()
+                    : (index == size - 1) ? unlinkLast()
+                    : unlink(searchNodeByIndex(index));
```

Comment:
This ternary operator is very hard to read. Please play with code formatting or use if/else structure